src/main/java/no/nav/foreldrepenger/oversikt/aareg/AktivitetskravArbeidDokumentasjonsKravTjeneste.java

```diff
@@ -0,0 +1,135 @@
+
+    private static final BigDecimal KRAV_FOR_DOKUMENTASJON = BigDecimal.valueOf(75);
+
+    private ArbeidsforholdTjeneste arbeidsforholdTjeneste;
+
+    public AktivitetskravArbeidDokumentasjonsKravTjeneste() {
+        //CDI
+    }
+
+    @Inject
+    public AktivitetskravArbeidDokumentasjonsKravTjeneste(ArbeidsforholdTjeneste arbeidsforholdTjeneste) {
+        this.arbeidsforholdTjeneste = arbeidsforholdTjeneste;
+    }
+
+    public boolean krevesDokumentasjonForAktivitetskravArbeid(PerioderMedAktivitetskravArbeid request) {
+
+        // Lager en tidslinje som inneholder periodene som skal sjekkes med hensyn på behov for dokumenasjon av aktivitetskrav = arbeid.
+        var requestTidslinje = request.aktivitetskravPerioder().stream()
+            .map(p -> new LocalDateSegment<>(p, BigDecimal.ZERO))
+            .collect(Collectors.collectingAndThen(Collectors.toList(), l -> new LocalDateTimeline<>(l, bigDesimalSum())));
+
+        // Ingen perioder i request = ikke behov for dokumentasjon
+        if (requestTidslinje.isEmpty()) {
+            return false;
+        }
+
+        // Slår opp i Aa-register, velger typer arbeidsforhold som er relevante, mapper om og grupperer på arbeidsgiver
+        var ident = request.morFødselsnummer();
+        var morsAktivitet = arbeidsforholdTjeneste.finnAlleArbeidsforholdForIdentIPerioden(ident, requestTidslinje.getMinLocalDate(), requestTidslinje.getMaxLocalDate())
+            .stream()
+            .filter(a -> RELEVANT_ARBEID.contains(a.arbeidsforholdIdentifikator().arbeidType()))
+            .collect(Collectors.groupingBy(this::aktivitetskravNøkkel));
+
+        // Ikke funnet noe arbeid, dokumentasjon er nødvendig
+        if (morsAktivitet.isEmpty()) {
+            return true;
+        }
+
+        return finnesBehovForDokumentasjon(requestTidslinje, morsAktivitet);
+    }
+
+    private boolean finnesBehovForDokumentasjon(LocalDateTimeline<BigDecimal> requestTidslinje,
+                                                Map<String, List<Arbeidsforhold>> morsAktivitet) {
+        Map<String, LocalDateTimeline<BigDecimal>> tidslinjerPrOrgnummer = new LinkedHashMap<>();
+
+        var requestSpanTidslinje = new LocalDateTimeline<>(requestTidslinje.getMinLocalDate(), requestTidslinje.getMaxLocalDate(), BigDecimal.ZERO);
+
+        // Koden nedenfor kunne vært enda mer forenklet men vi venter på litt mer erfaring med data fra Aa-register
+        // Vi skal potensielt se bort fra enkelte typer permisjon eller permisjon for tilfelle med stillingsprosent 0%
+        for (var entry : morsAktivitet.entrySet()) {
+            var orgnr = entry.getKey();
+            var arbeidsforhold = entry.getValue();
+            // Sjekker om det finnes permisjoner i noen av de søkte periodene. Denne vil ha behov for tuning etterhvert (fx stilling 0% med permisjon).
+            var permisjonProsentTidslinje = permisjonTidslinje(arbeidsforhold);
```

Comment:
Kanskje man kan ha en sjekk først som går gjennom alle arbeidsforhold og sjekker om det er noe permisjoner i intervallet => Trenger dokumentasjon return true.

Deretter gjør vi logikken om "Lager tidslinje med stillingsprosent fra Aa-register, fyller på med 0% for perioder uten arbeid og beskjærer mot request." og summerer opp stillingsprosentene?

Nå føles det litt roretete ut å ha begge tilfellene i samme for loop.

Reply:
Men det er godt mulig vi får logikk pr arbeidsgiver. 
Delvis fixed